Helm install via yaml and node management

Himanshukrojha opened the pull request:
- Helm Chart Installation via YAML:

Adds support for installing Helm charts using YAML manifests, improving flexibility for Kubernetes resource management.

- Node Management Enhancements:

Updates and refines node management capabilities, including improved tools for node operations and better integration with the unified kubectl command set.

-fixed top command usage

Flux159 commented on src/tools/helm-template-apply.ts:
```diff
@@ -0,0 +1,174 @@
+
+export const helmTemplateApplySchema = {
+  name: "helm_template_apply",
+  description: "Install a Helm chart using template generation and kubectl apply (bypasses authentication issues)",
```
I'm wondering if this description should be a bit more descriptive for an LLM - "Install a Helm chart using template generation and kubectl apply. Attempt to use this if helm_install has authentication issues or errors".

The other thing is if it would be better to replace the regular helm_install instead of having a new tool that essentially does the same thing?

rr-paras-patel replied:
+1

Flux159 commented on README.md:
```diff
@@ -206,8 +212,254 @@ The following destructive operations are disabled:
+- `node_management`: Node management operations (can drain nodes)
 - `kubectl_generic`: General kubectl command access (may include destructive operations)
 
+### Helm Template Apply Tool
```
I feel like this should be in some docs/helm_install.md and docs/cleanup_pods.md file (readme is already pretty log & describing every tool in the readme itself might be excessive)

Flux159 reviewed (changes requested):
Hi, I think that this is a good PR - requesting changes for 2 reasons:
* It needs tests for the new tools
* There's some discussion we should have around helm_template_install specifically because it's kinda a duplicate tool to fix errors in regular helm_install?

CC @rr-paras-patel - any comments around the cleanup & node_management tools?

rr-paras-patel commented on README.md:
```diff
@@ -206,8 +212,254 @@ The following destructive operations are disabled:
+
+#### Usage Examples
+
+**1. List all nodes:**
```
existing `kubectl_get` cover this operation.

rr-paras-patel commented on src/index.ts:
```diff
@@ -418,6 +440,60 @@ server.setRequestHandler(
+          );
+        }
+
+        case "cleanup_pods": {
```
existing `kubectl_delete` method should cover this. and LLM should be able to pass required params.

Himanshukrojha commented:
Hi @Flux159 and @rr-paras-patel ,
Thanks for the review and feedback , I will make the required changes.

FYI for : "here's some discussion we should have around helm_template_install specifically because it's kinda a duplicate tool to fix errors in regular helm_install?"
- It removes dependency of kubeconfig api version mismatch when installing some helm charts , so should I merge this new helm installation via yaml step into old one ?

Himanshukrojha commented:
> please address comments


Hi @rr-paras-patel ,
Now I have , Merged helm-template-apply into helm-operations.ts as optional parameter.
Created separate documentation files and moved detailed docs out of README.
Removed redundant "list" operation from node-management tool.
Deleted cleanup_pods tool entirely, functionality covered by kubectl_get + kubectl_delete also updated readme regarding it.

Himanshukrojha commented:
Hi @rr-paras-patel @Flux159 , Please review.

Himanshukrojha commented on src/tools/helm-operations.ts:
```diff
@@ -89,158 +129,337 @@ export const uninstallHelmChartSchema = {
+ * @param params - Uninstall parameters
+ * @returns Promise with uninstall result
+ */
+export async function uninstallHelmChart(params: {
+  name: string;
+  namespace: string;
+}): Promise<{ content: { type: string; text: string }[] }> {
```
Hi @Flux159 , I removed HelmOperation type because it only had name and namespace, but different helm operations need different parameters so I used unnamed type to give each function exactly the parameters it needs.

Himanshukrojha replied:
Yes , sure I will keep the named types and make them specific.

Flux159 replied:
Perfect thanks! will give @rr-paras-patel some time to take a look, otherwise will merge & make a new release tomorrow.

Himanshukrojha replied:
Sure , thanks @Flux159 !
